Thorfile

```diff
@@ -1,16 +1,58 @@
+require 'yaml'
+require 'fileutils'
+require 'tmpdir'
+require 'digest'
 require 'thor/group'
 
 module Middleman
   class Generator < ::Thor::Group
     include ::Thor::Actions
 
-    source_root File.expand_path(File.dirname(__FILE__))
+    TEMPLATE_VERSION_FILE = '.template_version'.freeze
+
+    class_option 'template',
+                 aliases: '-T',
+                 default: 'alphagov/tech-docs-template'
+
+    source_root __dir__
 
     def detect_if_first_time_install
-      if option_set?('FIRST_TIME')
-        @first_time = parse_boolean('FIRST_TIME')
-      else
-        @first_time = yes?('Are you creating a completely new documentation project?')
+      @first_time = option_set?('FIRST_TIME') || !File.exist?('config.rb')
+    end
+
+    def clone_existing_version
+      return if @first_time || !File.exist?(TEMPLATE_VERSION_FILE)
+      template = YAML.load_file(TEMPLATE_VERSION_FILE)
```

Comment:
Do we need to do anything to sanity check the template version file? Thinking if e.g. there's a git conflict in that file and it can't be parsed as YAML, or for some reason it's empty, etc, we should try and display a useful error message.